src/cc_catalog_airflow/dags/provider_api_scripts/walters_art_museum.py

```diff
@@ -0,0 +1,209 @@
+"""
+Content Provider:       Walters Art Museum
+
+ETL Process:            Use the API to identify all CC licensed images.
+
+Output:                 TSV file containing the images and the
+                        respective meta-data.
+
+Notes:                  http://api.thewalters.org/
+                        Rate limit: 250000 Per Day Per Key
+"""
+
+import os
+import logging
+import common.requester as requester
+import common.storage.image as image
+from util.loader import provider_details as prov
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.DEBUG
+)
+
+logger = logging.getLogger(__name__)
+
+DELAY = 1
+LIMIT = 250000
+PROVIDER = prov.WALTERS_DEFAULT_PROVIDER
+REQUEST_TYPE = 'objects'
+ENDPOINT = f'http://api.thewalters.org/v1/{REQUEST_TYPE}'
+API_KEY = os.getenv('WALTERS_ART_MUSEUEM_KEY')
+MUSEUM_SITE = "https://art.thewalters.org"
+LICENSE = 'CC0 1.0'
+
+# The API takes api_key as a query and not as a header
+DEFAULT_QUERY_PARAMS = {
+    'accept': 'json',
+    'pageSize': 100,
+    'orderBy': 'classification'
+}
+
+QUERY_CLASSIFICATION = [
+    'Miniatures',
+    'Stained & Painted Glass',
+    'Lacquer & Inlay',
+    'Ceramics',
+    'Precious Stones & Gems',
+    'Pearl, Horn, Coral & Shell',
+    'Sculpture',
+    'Textiles',
+    'Painting & Drawing',
+    'Prints',
+    'Coins & Medals',
+    'Arms & Armor',
+    'Mosaics & Cosmati',
+    'Niello',
+    'Paper & Paper-Mache',
+    'Wood',
+    'Enamels',
+    'Manuscripts & Rare Books',
+    'Ivory & Bone',
+    'Mummies & Cartonnage',
+    'Timepieces, Clocks & Watches',
+    'Glasswares',
+    'Metal',
+    'Gold, Silver & Jewelry',
+    'Stone',
+    'Resin, Wax & Composite',
+    'Leather'
+]
+
+delayed_requester = requester.DelayedRequester(DELAY)
+image_store = image.ImageStore(provider=PROVIDER)
+
+
+def main():
+
+    logger.info("Begin: Walters Art Museum provider script.")
+
+    for class_param in QUERY_CLASSIFICATION:
+        logger.info(f"Obtaining Images of Classfication: {class_param}")
+        image_list = _get_image_list(class_param)
+        _ = _process_image_list(image_list)
+
+    total_images = image_store.commit()
+    logger.info(f"Total Images: {total_images}")
+    logger.info("Terminating Script!")
+
+
+def _get_image_list(
+        class_param,
+        endpoint=ENDPOINT,
+        retries=5
+):
+    image_list = []
+    cond = True
+    while cond:
+        query_params = _build_query_param(class_param)
+        json_response_inpydict_form = delayed_requester.get_response_json(
+            endpoint=endpoint,
+            retries=retries,
+            query_params=query_params,
+        )
+
+        items_list = _extract_items_list_from_json(json_response_inpydict_form)
+        if items_list is None:
+            break
+        for img in items_list:
+            image_list.append(img)
+
+        if json_response_inpydict_form.get("NextPage") is False:

```

Comment:
This will loop indefinitely in the case where "NextPage" is missing from the JSON for any reason.  Please use a more robust check (it's better to break unnecessarily than send indefinite identical requests to a 3rd party API).
```suggestion
        if json_response_inpydict_form.get("NextPage") is not True:

```